src/store/flow/block.ts

```diff
@@ -49,13 +46,31 @@ export const mutations: MutationTree<IFlowsState> = {
     const block = findBlockOnActiveFlowWith(blockId, state as unknown as IContext)
     findBlockExitWith(exitId, block).test = value
   },
+  block_setExitConfigByPath(state, { exitId, blockId, path, value }: {exitId: string; blockId: string; path: string; value: object | string}) {
+    const block = findBlockOnActiveFlowWith(blockId, state as unknown as IContext)
+    set(findBlockExitWith(exitId, block).config, path, value)
+  },
   block_setExitSemanticLabel(state, { exitId, blockId, value }: { exitId: string, blockId: string, value: string }) {
     const block = findBlockOnActiveFlowWith(blockId, state as unknown as IContext)
     findBlockExitWith(exitId, block).semantic_label = value
   },
-  block_pushNewExit(state, { blockId, newExit }: {blockId: string; newExit: IBlockExit}) {
+  block_pushNewExit(state, { blockId, newExit, insertAtIndex = undefined, shouldUseCache = false }: {blockId: string; newExit: IBlockExit, insertAtIndex: number | undefined, shouldUseCache: boolean}) {
     const block = findBlockOnActiveFlowWith(blockId, state as unknown as IContext)
-    block.exits.push(newExit)
+    const blockExits = findBlockExitsRef(block, shouldUseCache)
+
+    if (!isNil(insertAtIndex)) { // insertion, eg: case for block types having branching exits option (Segregated | Unified)
+      blockExits.splice(insertAtIndex, 0, newExit);
+    } else { // just push
+      blockExits.push(newExit);
+    }
+  },
+  block_updateExits(state, { block, newExits, shouldUseCache = false }: { block: IBlock; newExits: IBlockExit[]; shouldUseCache: boolean }) {
+    if (shouldUseCache) {
+      // @ts-ignore: TODO: remove this once IBlock has vendor_metadata key
+      set(block.vendor_metadata, 'io_viamo.cache.outputBranching.segregatedExits', newExits);
```

Comment:
[] @bzabos will take time to look at the PR
[] @safydy-r  will re-confirm with Jon if select many really needs segregated exits

Reply:
@safydy For future reference: via `voto5` this feature is used in:
![image](https://user-images.githubusercontent.com/58382/116595434-8b693280-a8e0-11eb-8688-3ce9d253b11a.png)

Reply:
Jon confirmed in Jira ticket, that always using `Unified exits` for select many would make the most sense

Reply:
@bzabos , about the voto5 reference, I already copied this component from voto5 and adapted it for flow-build (store, props, etc).